### ISSUE TITLE
Make new domain invitations expire after a period of time

### ISSUE THREAD

**Issue by h-m-f-t** [cisagov/manage.get.gov#2993]
### Issue description

Domain invitations should be temporary, but right now they are long-lasting: they don't actually expire, and they exist until someone deletes, [cancels](https://github.com/cisagov/manage.get.gov/issues/1604), or retrieves them. Let's change that.

At least at launch, implementation shouldn't be retroactive; this issue shouldn't make outstanding invitations inert. We should communicate the change to the nature of invitations to users, make a campaign to get unmanaged domains managed, and then legacy cancel invites.

### Acceptance criteria

- [ ] Invitations have an "expired" state
- [ ] If someone is re-invited, the state should be updated and a new expiration date set (instead of creating a new invitation)
- [ ] All new invitations expire after 30 days (existing invitations should be left alone)
- [ ] Add a task to change state when the expiration date happens
- [ ] Content is updated in-registrar, in invitation emails, and on get.gov

### Additional context

If a user is the recipient of an invitation, and it has expired, we won't initially show them anything in the registrar. They just won't see the domain. If they have no domains, they'll already see the message "Don't see a domain you're expecting? Click for more info" - we can include expiration as one of the reasons there.

### Links to other issues

- ❌ blocked by #3552

**Comment by h-m-f-t** [comment]
We discussed this in team workshop; [notes](https://docs.google.com/document/d/1DLjvHcXA5LRK2NynAzoGbjtpbN757tTjD7f8_4GfJUg/edit?tab=t.0#heading=h.kqvaelh0y6iv).

**Comment by h-m-f-t** [comment]
* @katypies, this is an old issue, but it summarizes our invitation expiration intens well. I've added it to cisagov/.gov-roadmap#12 but I suspect it should just be closed. Can you review?

**Comment by katypies** [comment]
This is a good explanation, but yes, it's a duplicate of the work planned in #5192 (and the subsequent phases of the expiration work).
